src/datetime.rs

```diff
@@ -220,3 +233,27 @@ impl<T: chrono::TimeZone> From<chrono::DateTime<T>> for crate::DateTime {
         Self::from_chrono(x)
     }
 }
+
+/// Errors that can occur during [`DateTime`] construction and generation.
+#[derive(Clone, Debug)]
+#[non_exhaustive]
+pub enum Error {
```

Comment:
To allow users to refer to this type, we'll need to make the `datetime` module `pub` in `lib.rs`. Then users can say `bson::datetime::Result`.